fv3core/decorators.py

```diff
@@ -335,6 +335,32 @@ def decorator(func) -> FV3StencilObject:
         return decorator(definition)
 
 
+class FixedOriginStencil:
+    """Wrapped GT4Py stencil object explicitly genrating
+    and using the normalized origins."""
+
+    def __init__(self, func, origin, domain, **kwargs):
+        self.normalized_origin = gtscript.gt_definitions.normalize_origin_mapping(
+            origin
+        )
+        self.domain = domain
+        self.func = func
+        self.stencil_object = gtscript.stencil(
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
+            definition=self.func,
+            **kwargs,
+        )
+
+    def __call__(self, *args, **kwargs) -> None:
+        self.stencil_object(
+            *args,
+            **kwargs,
+            normalized_origin=self.normalized_origin,
+            domain=self.domain,
```

Comment:
It would be super nice to also include validate_args==global_config.get_validate_args() here and then remove the self.stencil_runtime_args assignment and use in calling stencils where we apply this. I know this is a little out of scope of your PR, but is a tedious silly thing that this object gives us the capability to remove

Reply:
this is done